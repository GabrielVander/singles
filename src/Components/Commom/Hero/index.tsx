import React, {Fragment} from "react";
import PropTypes from "prop-types";

import {Box, Heading, Paragraph} from "grommet";
import {Trans, useTranslation} from "react-i18next";

interface HeroProps {
    size: string
}

const Hero = ({size}: HeroProps) => {
    const {t} = useTranslation(['hero']);

    return (
        <Fragment>
            <Box direction="row">
                <Heading size={size}>{t('hero:heading')}</Heading>
            </Box>
            <Paragraph margin={{top: "none"}} textAlign="center" size="xxlarge">
                <Trans i18nKey={"hero:description"}>
                    Get beta access to one of the newest social media out there and connect with
                    your peers around the world with <b>Singles</b>
                </Trans>
            </Paragraph>
        </Fragment>
    );
};

Hero.propTypes = {
    size: PropTypes.string.isRequired
};

export default Hero;
