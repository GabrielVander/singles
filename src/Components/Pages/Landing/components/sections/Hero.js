import React from 'react';
import classNames from 'classnames';
import {SectionProps} from '../../utils/SectionProps';
import ButtonGroup from '../elements/ButtonGroup';
import Button from '../elements/Button';
import {Link} from "react-router-dom";

const propTypes = {
  ...SectionProps.types
}

const defaultProps = {
  ...SectionProps.defaults
}

const Hero = ({
                className,
                topOuterDivider,
                bottomOuterDivider,
                topDivider,
                bottomDivider,
                hasBgColor,
                invertColor,
                ...props
              }) => {

  const outerClasses = classNames(
    'hero section center-content',
    topOuterDivider && 'has-top-divider',
    bottomOuterDivider && 'has-bottom-divider',
    hasBgColor && 'has-bg-color',
    invertColor && 'invert-color',
    className
  );

  const innerClasses = classNames(
    'hero-inner section-inner',
    topDivider && 'has-top-divider',
    bottomDivider && 'has-bottom-divider'
  );

  return (
    <section
      {...props}
      className={outerClasses}
    >
      <div className="container-sm">
        <div className={innerClasses}>
          <div className="hero-content">
            <h1 className="mt-0 mb-16 reveal-from-bottom" data-reveal-delay="200">
              Great social network for <span className="text-color-primary">single parents</span>
            </h1>
            <div className="container-xs">
              <p className="m-0 mb-32 reveal-from-bottom" data-reveal-delay="400">
                We seek to provide a safe social environment where single parents can connect with each other, share
                common interests, experiences, advice and tips on all stages of raising children (and who knows, maybe
                even love <span role="img" aria-label="winking emoji">ðŸ˜‰</span>)
              </p>
              <div className="reveal-from-bottom" data-reveal-delay="600">
                <ButtonGroup>
                  <Link to="/register">
                    <Button tag="a" color="primary" wideMobile>
                      Get started
                    </Button>
                  </Link>
                </ButtonGroup>
              </div>
            </div>
          </div>
        </div>
      </div>
    </section>
  );
}

Hero.propTypes = propTypes;
Hero.defaultProps = defaultProps;

export default Hero;
