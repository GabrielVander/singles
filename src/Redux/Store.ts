import {createStore} from "redux";
import rootReducer from "./Reducers";

class Store {
    static INITIAL_GLOBAL_STATE = {};
    // @ts-ignore
    static store = createStore(rootReducer, Store.INITIAL_GLOBAL_STATE, window.__REDUX_DEVTOOLS_EXTENSION__ && window.__REDUX_DEVTOOLS_EXTENSION__());
}

export default Store;
