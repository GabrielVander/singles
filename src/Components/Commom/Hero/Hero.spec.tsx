import {render} from "@testing-library/react";
import {Provider} from "react-redux";
import Store from "../../../Redux/Store";
import {BrowserRouter} from "react-router-dom";
import React from "react";
import Hero from "./index";

test('matches snapshot', () => {
    const component = render(
        <Provider store={Store.store}>
            <BrowserRouter>
                <Hero size="small"/>
            </BrowserRouter>
        </Provider>
    );
    expect(component).toMatchSnapshot();
});
