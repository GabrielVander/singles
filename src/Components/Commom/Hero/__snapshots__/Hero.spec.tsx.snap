// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="StyledBox-sc-13pk1d4-0 bEnuwL"
      >
        <h1
          class="StyledHeading-sc-1rdh4aw-0 iqLQKP"
        >
          hero:heading
        </h1>
      </div>
      <p
        class="StyledParagraph-tbetod-0 eyLdxA"
      >
        Get beta access to one of the newest social media out there and connect with your peers around the world with
        <b>
          Singles
        </b>
      </p>
    </div>
  </body>,
  "container": <div>
    <div
      class="StyledBox-sc-13pk1d4-0 bEnuwL"
    >
      <h1
        class="StyledHeading-sc-1rdh4aw-0 iqLQKP"
      >
        hero:heading
      </h1>
    </div>
    <p
      class="StyledParagraph-tbetod-0 eyLdxA"
    >
      Get beta access to one of the newest social media out there and connect with your peers around the world with
      <b>
        Singles
      </b>
    </p>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
